File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/error/SignUpValidationError.java
package com.lattis.ellipse.domain.interactor.error;

import java.util.ArrayList;
import java.util.List;

public class SignUpValidationError extends Throwable {

    private List<Status> status = new ArrayList<>();

    public SignUpValidationError(List<Status> status) {
        this.status = status;
    }

    public List<Status> getStatus() {
        return status;
    }

    public enum Status{
        INVALID_EMAIL,
        INVALID_PHONE_NUMBER,
        INVALID_PASSWORD
    }
}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/error/SignUpValidationError.kt
package com.lattis.domain.error

import java.util.ArrayList

class SignUpValidationError(status: List<Status>) : Throwable() {

    var status: List<Status> = ArrayList()

    init {
        this.status = status
    }

    enum class Status {
        INVALID_EMAIL,
        INVALID_PASSWORD
    }
}
